g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java

```diff
@@ -152,17 +152,11 @@
 		}
 
 		if (categoryList.contains(ConstantsKeys.PLURALS)) {
-			Map<String, Object> pluralPatternMap = getPattern(language, Arrays.asList(ConstantsKeys.PLURALS));
-			Map<String, Object> pluralRulesMap = new LinkedHashMap<>();
-			pluralRulesMap.put(ConstantsKeys.PLURAL_RULES, null);
-			categoriesMap.put(ConstantsKeys.PLURALS, pluralRulesMap);
-			patternMap.put(ConstantsKeys.IS_EXIST_PATTERN, true);
-			if (null != pluralPatternMap.get(ConstantsKeys.CATEGORIES)) {
-				Map<String, Object> pluralMap = (Map<String, Object>) pluralPatternMap.get(ConstantsKeys.CATEGORIES);
-				if (null != pluralMap.get(ConstantsKeys.PLURALS)) {
-					categoriesMap.put(ConstantsKeys.PLURALS, pluralMap.get(ConstantsKeys.PLURALS));
-				}
-			}
+			handleSpecialCategory(ConstantsKeys.PLURALS, language, categoriesMap);
+		}
+
+		if (categoryList.contains(ConstantsKeys.DATE_FIELDS)) {
+			handleSpecialCategory(ConstantsKeys.DATE_FIELDS, language, categoriesMap);
```

Comment:
The logic here will always use language to get 'dateFields' data, this may be not correct. 
When the combination of language and region is a valid locale, use locale(language&region) to get dateFields data; if invalid, use language to get data.

Reply:
fixed